### PR TITLE
feat: Publish an anonymous question. 

### LINKED ISSUE #162: [ADMIN] Publish  Questions
In the questions list, add the option to publish a question.

- Use the `is_public` to control this functionality 
- This options will be only available for admins 
- This options only will be visible for the questions that the value of `is_public` is `false`
- When an admin publishes a question, send an alert to the Slack channel, **ONLY WITH ANONYMOUS QUESTIONS.**
- For the visual part, you can use the following icons to represent the action:
    https://react-icons.github.io/react-icons/search?q=publish
    https://react-icons.github.io/react-icons/search?q=world
    https://react-icons.github.io/react-icons/search?q=eye

_Note: this option is similar to the pin and unpin questions and enabled and disabled questions_

### PR DESCRIPTION
**Pull request by GeorginaFraire**
#### What does this PR do?
- add the controllers to manage the new functionality to publish an anonymous question in the home page. 
- add the option in the QuestionRow Component to publish an anoymous question in the home page (this option is only available to admins) 

#### Where should the reviewer start?
- run the test (`npm run test:integration`)
- run the project and see the option in the home page (only if the user is admin) 

1. Login as normal user. (without admin permissions)
2. Verify that you cannot see any anonymous question. 
3. Log out.
4. Login as [admin](https://github.com/wizeline/wize-q-remix/wiki/How-to-give-admin-permissions-to-my-user).
5. Go to an anonymous question and publish it. Screen Shot 2023-01-26 at 10 37 47
6. Repeat the first step and verify that you can see the anonymous question that you published
7. :tada:

#### Any background context you want to provide?
The step BEFORE START is necessary, because at this moment when you create an anonymous question is created with the default value of `is_public`(`true`)
👀 👀  The option to publish a question is available if the value of `is_public` is false, once you publish a question the option will not appear and if the question is enabled the option it doesn't appear either.

#### What are the relevant tickets?
(Link to issues, related PR, JIRA issues, etc.)
Closes #162 

#### Screenshots
test: 
Screen Shot 2023-01-26 at 10 53 11

unpublish question:
Screen Shot 2023-01-26 at 10 54 59

question that has been published
Screen Shot 2023-01-26 at 10 55 48

disabled question
Screen Shot 2023-01-26 at 10 58 40

### CODE REVIEW

**Comment by albertcastaned** [comment]
For #143 we want to set `is_public` to true when creating anon questions right?

File: app/components/QuestionRow/QuestionRow.jsx [hunk 1]
```diff
@@ -131,21 +150,32 @@ function QuestionRow(props) {
...
+                  question
+                </Styled.ButtonTooltipMessage>
+                <Switch
+                  id={`question-${question.question_id}`}
+                  checked={question.is_enabled}
+                  onChange={handleStatusClick}
+                />
+              </Styled.DisableControls>
+              {(!question.is_public && question.is_enabled) && (
+                <Button
```
Comment (albertcastaned) [inline app/components/QuestionRow/QuestionRow.jsx:170]: Maybe we can add a tooltip to the button for clarity?

File: app/controllers/questions/publishQuestion.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { db } from 'app/utils/db.server';
+import { PIN_QUESTION_ERROR_MESSAGE, INVALID_PARAMS_FOR_OPERATION_ERROR_MESSAGE } from 'app/utils/constants';
+
+const publish = async (questionId) => {
```
Comment (albertcastaned) [inline app/controllers/questions/publishQuestion.js:4]: We could name the controller to something more clear like `setQuestionAsPublic` or `publishQuestion`

**Comment by GeorginaFraire** [comment]
> For #143 we want to set `is_public` to true when creating anon questions right?

No, we set `is_public` to `false`, by default is `true`

Anonymous questions: is_public = false.
Non-anonymous questions: is_public = true.

File: app/controllers/questions/publishQuestion.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    });
+
+    return {
+      successMessage: 'The questions has been published',
+      question: updatedQuestion,
+    };
+  } catch (error) {
+    return {
+      error: {
+        message: 'publishQuestiosn.js error',
```
Comment (albertcastaned) [inline app/controllers/questions/publishQuestion.js:28]: Do we want to show this to the user? Maybe we can show a generic error

**Review by albertcastaned — COMMENTED** [review]
Left some minor comments but other than that looks good to me 👍

**Comment by albertcastaned** [comment]
> > For #143 we want to set `is_public` to true when creating anon questions right?
> 
> No, we set `is_public` to `false`, by default is `true`
> 
> Anonymous questions: `is_public` = `false`. Non-anonymous questions: `is_public` = `true`.

Yes sorry that's what I meant. Also saw that it was already written in the ticket. I will update my branch with this one to make that change 😄

File: app/controllers/questions/publishQuestion.js (hunk 3 above)
Comment (GeorginaFraire) [reply app/controllers/questions/publishQuestion.js]: Done ✅

File: app/controllers/questions/publishQuestion.js (hunk 2 above)
Comment (GeorginaFraire) [reply app/controllers/questions/publishQuestion.js]: Done ✅

File: app/components/QuestionRow/QuestionRow.jsx (hunk 1 above)
Comment (GeorginaFraire) [reply app/components/QuestionRow/QuestionRow.jsx]: Done ✅

**Comment by GeorginaFraire** [comment]
Tooltip:

Screen Shot 2023-01-26 at 13 31 57

**Review by albertcastaned — APPROVED** [review]
💯

**Review by Hoosep — APPROVED** [review]

**Pull request merged**